webapp-django/crashstats/crashstats/static/crashstats/js/socorro/exploitability_report.js

```diff
@@ -0,0 +1,119 @@
+                .addClass('impossible');
+        }
+
+        // the "Next" link
+        if (page * size < count) {
+            $('a.next', parent)
+                .attr('href', '#' + (page + 1))
+                .removeClass('impossible');
+        } else {
+            $('a.next', parent)
+                .attr('href', '#' + page)
+                .addClass('impossible');
+        }
+
+        // The "Last" link
+        if (page * size < count) {
+            $('a.last', parent)
+                .attr('href', '#' + Math.ceil(count / size))
+                .removeClass('impossible');
+        } else {
+            $('a.last', parent)
+                .attr('href', '#' + Math.ceil(count / size))
+                .addClass('impossible');
+        }
+    }
+
+    function fetchReports(page) {
+        var url = document.location.pathname + document.location.search;
+        url += '&format=json';
+
+        return $.getJSON(url, {page: page})
+        .done(function(response) {
+            var tbody = $('table.data-table tbody');
+            $('tr', tbody).remove();
+            $.each(response.crashes, function(i, crash) {
+                var tr = $('<tr>');
+                tr.append(
+                    $('<td>').append(
+                        $('<a>')
+                        .attr('href', crash.url)
+                        .text(crash.signature)
+                    )
+                );
+                var bugsHTML = '';
+                $.each(crash.bugs, function(i, bug) {
+                    bugsHTML += bug.html + ' ';
+                });
+                tr.append(
+                    $('<td>').addClass('bug_ids_more').html(bugsHTML)
+                );
+                tr.append($('<td>').text(crash.high_count));
+                tr.append($('<td>').text(crash.medium_count));
+                tr.append($('<td>').text(crash.low_count));
+                tr.append($('<td>').text(crash.none_count));
+                tbody.append(tr);
+            });
+            $('.pagination').show();
+            updatePaginationLinks(
+                page,
+                response.count,
+                response.batch_size
+            );
+            BugLinks.enhance();
+        })
+        .fail(function() {
+            console.error.apply(console, arguments);
+            alert("Loading reports failed.");
+        });
+    }
+
+    // onload
+    $(function() {
+
+        $('.pagination').on('click', 'a', function(e) {
+            if ($(this).hasClass('impossible')) {
+                e.preventDefault();
+            } else {
+                page = parseInt($(this).attr('href').replace('#', ''), 10);
+                fetchReports(page);
+            }
+        });
+
+        // let's the first piece of data going
+        var page = 1;
+        if (document.location.hash) {
+            page = parseInt(document.location.hash.replace('#', ''), 10);
+        }
+        fetchReports(page);
+    });
+
+}($, document));
```

Comment:
This whole file seems out-of-place in the current Socorro. Ideally we will rewrite Socorro as a JS app some day, but at the moment it's a django / jinja app, and it is weird to me that we mix JS and jinja as templating tools. 

Also, understanding an HTML structure built with jQuery is a horrible experience to me. The day we do JS templating, I really hope we'll use something like JSX.